application/Commands/Types/Relationship.php

```diff
@@ -328,4 +328,68 @@ protected function delete_item( $slug, $do_cleanup = true ) {
 
 		return true;
 	}
-}
+
+	/**
+	 * Displays posts related to a given post.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --post=<number>
+	 * : The ID of the post.
+	 *
+	 *	--relationship=<string>
+	 * : The relationship slug.
+	 *
+	 * [--role-to-return=<string>]
+	 * : The role to return.
+	 *
+	 * [--query-by-role=<string>]
+	 * : The role to query by.
+	 *
+	 * [--format=<format>]
+	 * : The format of the output. Can take values: table, csv, json, count, yaml. Default: table.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *    wp types relationship get-related-posts --post=123 --relationship=slug
+	 *    wp types relationship get-related-posts --post=123 --relationship=slug --role-to-return=child --query-by-role=parent
+	 *    wp types relationship get-related-posts --post=123 --relationship=slug --role-to-return=child --query-by-role=parent --format=json
+	 *
+	 * @subcommand get-related-posts
+	 * @synopsis --post=<number> --relationship=<string> [--role-to-return=<string>] [--query-by-role=<string>] [--format=<format>]
+	 * @link https://toolset.com/documentation/customizing-sites-using-php/post-relationships-api/#toolset_get_related_posts
+	 */
+	public function get_related_posts( $args, $assoc_args ) {
+		$defaults    = array (
+			'role-to-return' => 'child',
+			'query-by-role' => 'parent',
+			'format' => 'table',
+		);
+		$assoc_args = wp_parse_args( $assoc_args, $defaults );
+		$relationship_slug = $assoc_args['relationship'];
+		$post = $assoc_args['post'] ;
+
+		$query_args = [
+			'orderby' => 'title',
+			'order' => 'ASC',
+			'return' => 'post_id',
+			'limit' => 999,
+			] ;
+		$query_args['role_to_return'] = $assoc_args['role-to-return'];
+		$query_args['query_by_role'] = $assoc_args['query-by-role'];
+
+		$related_posts = toolset_get_related_posts ( $post, $relationship_slug, $query_args ) ;
+
+		if ( count($related_posts) > 0 ) {
+			$columns = [ 'ID', 'post_title',] ;
+			$items = array_map ( function ( $related_post ) {
+				return [
+					'ID' => $related_post,
+					'post_title' => get_the_title ($related_post)
+				] ;
+			}, $related_posts) ;
+
+			\WP_CLI\Utils\format_items( $assoc_args['format'], $items, $columns );
+		}
```

Comment:
Maybe also print something if there are no results?

Reply:
I thought about this but didn't know what content to output. A WP CLI error didn't seem right since zero posts might not be an error. Some WP CLI commands just output table headers, which seems odd to me. For example, here's what `wp plugin list --update=available` outputs when all plugins are up-to-date:

```
+------+--------+--------+---------+
| name | status | update | version |
+------+--------+--------+---------+
+------+--------+--------+---------+
```

Did you have any suggestions?

Reply:
@baizmandesign I agree, an error or an empty table don't feel right. How about simply printing "No results found"?

Then again, that might cause problems for certain formats (if someone needs to process the output programmatically). So perhaps just always printing the result using `\WP_CLI\Utils\format_items`, wherever the result is an empty array or not, will be the best approach. But a completely empty result might also be seen as an error.

Maybe there is some official WP CLI guidance on this? I'm not sure...

Reply:
I was looking, for reference, at the output of some other commands, such as `wp post list`. When no posts exist, the command returns a table header and nothing else:

```
+----+------------+-----------+-----------+-------------+
| ID | post_title | post_name | post_date | post_status |
+----+------------+-----------+-----------+-------------+
+----+------------+-----------+-----------+-------------+
```

That seems to be the default display, so I guess we should go with that.

Reply:
Perfect, thanks a lot for the research.